src/shipit_static_analysis/shipit_static_analysis/clang.py

```diff
@@ -43,6 +44,27 @@
     '--from-build', 'linux64-clang-tidy'
 ]
 
+__third_party = None
+
+
+def list_third_party(work_dir, path='tools/rewriting/ThirdPartyPaths.txt'):
+    '''
+    List third party directories using mozilla-central file
+    Uses a local cache to avoid reading every time the file
```

Comment:
I think you are missing the runtime context here. The local cache is only in memory, so it's renewed for every new task execution.
I agree there is not really a problem reading the file every time, but small optimizations can go a long way.